### DISCUSSION TITLE
Can't join the round - game keeps downloading save file from server

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#14995] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

https://github.com/user-attachments/assets/2a2a71e3-79a1-4085-a04b-be1dc3e55624

It happens regularly when we join the heavy modded dedicated server
player numbers, mod numbers, network connection can contribute to this issue

### Reproduction steps

1. start a multiplayer campaign
2. let other players connect the server?

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Multiplayer hosted using a dedicated server

### Version

v1.6.18.1 (Unto the Breach Update)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Error in ClientRead. Owned submarine index was out of bounds (list of server submarines is empty).
```

**Comment by Regalis11** [comment]
Thank you for the report (and sorry for taking so long to get back to you)! 

We haven't yet been able to reproduce the issue, but we suspect it might have something to do with a known issue that causes the server to sometimes "spam" the server settings to the client, causing the lobby to constantly refresh. This is caused by a bug in the PvP outpost dropdown: when a game mode other than PvP is selected, clients will keep attempting to re-select an outpost in the (hidden) dropdown and cause the server to spam settings updates. This issue has been fixed in the unstable build, but the fix isn't out in the public version yet.

If you have access to the server console, any details on what appears there would most likely help us a lot in diagnosing the issue. If you can see the output, do you see any kind of error messages, warnings or unusual message spam there?

In any case, I'm sorry for the inconvenience, and I hope we can figure out what's causing this asap.

**Reply by A2ur3** [reply]
no error on server side
sometimes client will receive this message
![20241115010647](https://github.com/user-attachments/assets/37b17254-1ebd-4167-922e-ff60f2d00d0c)
and sometimes game won't download save file

**Reply by 4-S-H** [reply]
![image](https://github.com/user-attachments/assets/68d3a0d3-6281-4415-a38e-68713235ce39)
This happens on client side console when it happens to me

**Reply by 4-S-H** [reply]
and this
![image](https://github.com/user-attachments/assets/0f17d522-52fe-45ca-9299-8fb5f8917ee9)

**Comment by Jasontti** [comment]
Tested a potential fix for this issue.
Impossible to reproduce the original issue, but didn't see any issues after several tries.
Fix is coming in the winter update.
Closing discussion as resolved for now, but feel free to reopen if new information comes up.

**Reply by Regalis11** [reply]
I think we could maybe keep this report open for now, I'm not 100% sure this issue was caused by the bug we fixed. For context, here's what we fixed:

> Fixed an issue that could cause clients to get stuck in an empty lobby. This could potentially explain FakeFishGames/Barotrauma#15143, especially since there's a known issue in the public build that can cause the server to spam lobby updates and cause the lobby update ID to go up rapidly.
> 
> Happened when the lobby screen's update ID had gone up to a very large value (> 32 767, half-way to wrapping around). In that situation, the client's `LastRecvLobbyUpdate` would be considered higher than the server's update ID (as if it had wrapped around to 0), and the server would not send any lobby updates.

**Reply by 4-S-H** [reply]
I was stuck in an empty lobby at the time so at least it will fix something.
